### DISCUSSION TITLE
Why do we use client.WatchAsync<TEntity>() rather than client.Watch<TEntity>() for the entity watcher?

### DISCUSSION THREAD

**Discussion by slacki123** [dotnet/dotnet-operator-sdk#795] [Q&A]
Hi, just wondering why do we use the `WatchAsync<TEntity>()` instead of the `Watch<TEntity>()` method to watch for custom resources here
https://github.com/buehler/dotnet-operator-sdk/blob/main/src/KubeOps.Operator/Watcher/ResourceWatcher%7BTEntity%7D.cs#L139

I recently implemented some of my own watchers with the Watch<TEntity>() method and it works really well with simple implementation. What's the reasoning for using `WatchAsync<TEntity>()`? Is it because `Watch<TEntity>()` wasn't available when the operator was first written or is it lacking some functionality?

**Comment by buehler** [comment]
Hey @slacki123 

The reason is simply that when I rewrote v8, I used async enumerables and other modern language features. There is no particular reason.
